engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -89,17 +89,17 @@ def self.verify_instance(request, logger, system)
     base_product = system.products.find_by(product_type: 'base')
     return false unless base_product
 
-    instance_data = request.headers['X-Instance-Data'].to_s
+    decoded_instance_data = Base64.decode64(request.headers['X-Instance-Data'].to_s)
```

Comment:
Are we missing a test here that sends base64 encoded instance data?

Reply:
yes